File: mobile/src/main/java/com/siliconlabs/bledemo/Browser/Model/Logs/ServicesDiscoveredLog.java
package com.siliconlabs.bledemo.Browser.Model.Logs;

import android.bluetooth.BluetoothGatt;

import com.siliconlabs.bledemo.Browser.Model.LogType;

public class ServicesDiscoveredLog extends Log {

    public ServicesDiscoveredLog(BluetoothGatt gatt, final int status) {
        setLogTime(getTime());
        setLogInfo(gatt.getDevice().getAddress() + "(" + getDeviceName(gatt.getDevice().getName()) + "): "
                + parseStatus(status));
        setLogType(LogType.INFO); //malo wazne
        setDeviceAddress(gatt.getDevice().getAddress());
    }

    private static String parseStatus(int status) {
        if (status == BluetoothGatt.GATT_SUCCESS) {
            return "Successfully discovered services";
        }
        return "Unsuccessfully discovered services with status: " + status;
    }
}

File: mobile/src/main/java/com/siliconlabs/bledemo/Browser/Models/Logs/ServicesDiscoveredLog.kt
package com.siliconlabs.bledemo.Browser.Models.Logs

import android.bluetooth.BluetoothGatt
import com.siliconlabs.bledemo.Browser.Models.LogType

class ServicesDiscoveredLog(gatt: BluetoothGatt, status: Int) : Log() {
    companion object {
        private fun parseStatus(status: Int): String {
            return if (status == BluetoothGatt.GATT_SUCCESS) {
                "Successfully discovered services"
            } else "Unsuccessfully discovered services with status: $status"
        }
    }

    init {
        logTime = getTime()
        logInfo = (gatt.device.address + " (" + getDeviceName(gatt.device.name) + "): "
                + parseStatus(status))
        logType = LogType.INFO
        deviceAddress = gatt.device.address
    }
}